02/app.js

```diff
@@ -0,0 +1,104 @@
+const formEl = document.querySelector("form");
+const PASSWORD_LENGTH = 4;
+
+const regexEmail =
+  /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
+
+formEl.addEventListener("submit", handleSubmit);
+
+const validations = {
+  login: { validation: validateEmail, errorMsg: "Email is not valid" },
+  pass1: {
+    validation: checkPassword,
+    errorMsg: `password must have ${PASSWORD_LENGTH} characters or more`,
+  },
+  pass2: {
+    validation: checkMatchingPasswords,
+    errorMsg: "passwords do not match",
+  },
+  accept: {
+    validation: validateCheckbox,
+    errorMsg: "regulations must be accepted",
+  },
+};
+
+function handleSubmit(e) {
+  const errors = [];
+  const self = e.target;
+  const inputs = self.elements;
+  checkInputs(inputs, validations, errors);
+
+  checkErrors(e, errors);
+  if (errors.length === 0) {
+    Array.from(formEl.elements).forEach((el) => (el.value = ""));
+  }
+}
+
+function checkInputs(inputs, validations, errorArr) {
+  for (const element of inputs) {
+    const keys = Object.keys(validations);
+    if (keys.includes(element.name)) {
+      // nie wiem jak zrobić to zgrabniej
+      // problem z tym, że pass2 musi mieć dwa argumenty
+      // może przekazywać argumenty w obiekcie?
+      let isElValid;
+      if (element.name === "pass2") {
+        isElValid = validations[element.name].validation(element, inputs.pass1);
+      } else {
```

Comment:
Możesz zrobić tak, że w `validations.pass2` definiujesz nową opcję tj. `compareWith` i tam podajesz nazwę klucza dla którego ma być porównana wartość :) Wtedy to będzie bardziej elastyczne.